lib/pdf/core/page.rb

```diff
@@ -22,6 +22,26 @@ def initialize(document, options = {})
         end
       end
 
+      # Prepend a content stream containing 'q',
+      # and append a content stream containing 'Q'.
+      # This ensures that prawn has a pristine graphics state
+      # before it starts adding content.
+      def wrap_graphics_state
+        unless dictionary.data[:Contents].is_a?(Array)
+          dictionary.data[:Contents] = [content]
+        end
```

Comment:
This whole conditional can be written like this:

```ruby
dictionary.data[:Contents] = Array(dictionary.data[:Contents])
```

It will do essentially the same.

Reply:
Good point, updated